File: planner/app/src/main/java/com/sunilson/firenote/baseClasses/Checklist.java
package com.sunilson.firenote.baseClasses;

import java.util.HashMap;
import java.util.Map;

/**
 * @author Linus Weiss
 */

public class Checklist extends Element {

    private Map<String, ChecklistElement> elements = new HashMap<>();

    public Checklist() {
        super();
    }

    public Checklist(String noteType, String title) {
        super(noteType, title);
    }

    public Map<String, ChecklistElement> getElements() {
        return elements;
    }

    public void setElements(Map<String, ChecklistElement> elements) {
        this.elements = elements;
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/data/models/Checklist.kt
package com.sunilson.firenote.data.models

import java.util.*
import kotlin.collections.HashMap

class Checklist( var elements: HashMap<String, ChecklistElement> = hashMapOf(),
                 elementID: String,
                 category: Category,
                 noteType: String,
                 color: Int,
                 locked: Boolean,
                 timestamp: Long = Date().time,
                 title: String = "New Element") : Element(elementID, category, noteType, color, locked, timestamp, title)